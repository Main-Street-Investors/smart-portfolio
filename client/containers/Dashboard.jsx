import React, { Component } from 'react';

class Dashboard extends Component {
  constructor(props) {
    super(props);
  }
  
  render() {
    return(
      <div>
        <h1>Dashboard</h1>
      </div>
    )
  }
}

export default Dashboard;
