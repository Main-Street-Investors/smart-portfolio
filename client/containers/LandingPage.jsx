import React, { Component } from 'react';

class LandingPage extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return(
      <div>
        <h1>Landing Page</h1>
      </div>
    )
  }
}

export default LandingPage;
