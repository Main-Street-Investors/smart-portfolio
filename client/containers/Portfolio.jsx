import React, { Component, useState } from 'react';
import { Link } from 'react-router-dom';
import {
  Switch,
  useRouteMatch,
  useParams,
  Route
} from "react-router-dom";
import Modal from 'react-bootstrap/Modal';
import Button from 'react-bootstrap/Button';
import Form from 'react-bootstrap/Form';


function Portfolio() {
  let match = useRouteMatch();

  const [showNewPortfolio, setShowNewPortfolio] = useState(false);
  const [showNewStock, setNewStock] = useState(false);
  const [showSellStock, setSellStock] = useState(false);


  const handleClose = () => setShowNewPortfolio(false);
  const handleClose2 = () => setNewStock(false);
  const handleClose3 = () => setSellStock(false);

  return (
    <div>
      <h1>Portfolio Page</h1>
      <Switch>
        <Route path={`${match.path}/:portfolioID`}>
          <Placeholder />
        </Route>
        <Route path={match.path}>
          <h3>Please select a portfolio.</h3>
        </Route>
      </Switch>
      <div>

      <Button>Portfolio 1</Button>
      <Button>Portfolio 2</Button>

      <Button>Create new Portfolio</Button>


      <Button><Link to="/Dashboard">Back to Dashboard</Link></Button>
      <Button><Link to="/LandingPage">Logout</Link></Button>


      <Button onClick={() => {
        setShowNewPortfolio(true);
      }}>Rename this Portfolio</Button>
      <Modal show={showNewPortfolio} onHide={handleClose}>
        <Modal.Header closeButton>
          <Modal.Title>Name your Portfolio</Modal.Title>
        </Modal.Header>
        <Form>
          <Form.Group>
            <Form.Label>Give your new portfolio a new name:</Form.Label>
            <Form.Control type="text" placeholder="Portfolio 1" />
          </Form.Group>
        </Form>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Close
          </Button>
          <Button variant="primary" onClick={handleClose}>
            Save Changes
          </Button>
        </Modal.Footer>
      </Modal>

      <Button onClick={() => {
        setNewStock(true);
      }}>Add Stock</Button>
      <Modal show={showNewStock} onHide={handleClose2}>
      <Modal.Header closeButton>
          <Modal.Title>New Stock</Modal.Title>
        </Modal.Header>
        <Form>
          <Form.Group>
            <Form.Label>Ticker Name:</Form.Label>
            <Form.Control type="text" placeholder="TSLA" />
            <Form.Label>Date Purchased:</Form.Label>
            <Form.Control type="date" placeholder="29-09-2020" />
            <Form.Label>Price Purchased:</Form.Label>
            <Form.Control type="number" min="1" placeholder="420"/>
            <Form.Label># of Shares:</Form.Label>
            <Form.Control type="number" min="1" placeholder="1"/>
            <Form.Label>Tags:</Form.Label>
            <Form.Control type="text" placeholder="tech"/>
          </Form.Group>
        </Form> 
        <Modal.Footer>
          <Button variant="primary" onClick={handleClose2}>
            Submit
          </Button>
        </Modal.Footer>
      </Modal>
      
      <Button onClick={() => {
        setSellStock(true);
      }}>Sell Stock</Button>
      <Modal show={showSellStock} onHide={handleClose3}>
      <Modal.Header closeButton>
          <Modal.Title>Sell Stock</Modal.Title>
        </Modal.Header>
        <Form>
          <Form.Group>
            <Form.Label>Ticker Name:</Form.Label>
            <Form.Control type="text" placeholder="TSLA" />
            <Form.Label>Date Purchased:</Form.Label>
            <Form.Control type="date" placeholder="29-09-2020" />
            <Form.Label>Price Purchased:</Form.Label>
            <Form.Control type="number" min="1" placeholder="420"/>
            <Form.Label># of Shares:</Form.Label>
            <Form.Control type="number" min="1" placeholder="1"/>
          </Form.Group>
        </Form> 
        <Modal.Footer>
          <Button variant="primary" onClick={handleClose3}>
            Submit
          </Button>
        </Modal.Footer>
      </Modal>

      <Button>View History</Button>

      </div>
    </div>
  );
}

function Placeholder() {
  let { portfolioID } = useParams();

  return (
    <div>Examining portfolio {portfolioID}</div>
  );
}

export default Portfolio;
