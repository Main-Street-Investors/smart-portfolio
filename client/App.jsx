import React, { Component } from 'react';
import LandingPage from './containers/LandingPage.jsx';
import Login from './containers/Login.jsx';
import Dashboard from './containers/Dashboard.jsx';
import Portfolio from './containers/Portfolio.jsx';
import {
  BrowserRouter,
  Switch,
  Route
} from "react-router-dom";
import Signup from './containers/Signup.jsx';

// Create App component
class App extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return(
      <BrowserRouter>
        <Switch>
          <Route path='/login' exact component={Login} />
          <Route path='/signup' exact component={Signup} />
          <Route path='/dashboard' exact component={Dashboard} />
          <Route path='/landingpage' exact component={LandingPage} />
          <Route path='/portfolio' component={Portfolio} />
          <Route path='/' exact component={LandingPage} />
          <Route path='/' render={() => <div>404</div>} />
        </Switch>
      </BrowserRouter>
    )
  }
}

export default App;
