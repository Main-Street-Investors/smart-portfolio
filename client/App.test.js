import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router';

import App from './App';
import LandingPage from './containers/LandingPage';
import Portfolio from './containers/Portfolio';
import Dashboard from './containers/Dashboard';

// react-router-dom has been mocked in __mocks__ folder, extra wrapping div present

describe('App', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/' ]}>
        <App />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('invalid path should redirect to 404', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/random' ]}>
        <App/>
      </MemoryRouter>
    );
    expect(wrapper.find(LandingPage)).toHaveLength(0);
  });

  it('renders the LandingPage', () => {
    const wrapper = mount(
      <MemoryRouter initialEntries={[ '/' ]}>
        <App />
      </MemoryRouter>
    );
    expect(wrapper.find(LandingPage).length).toEqual(1);
  });

});

describe('LandingPage', () => {
  test('snapshot renders', () => {
    const component = renderer.create(
      <MemoryRouter initialEntries={[ '/' ]}>
        <LandingPage />
      </MemoryRouter>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
