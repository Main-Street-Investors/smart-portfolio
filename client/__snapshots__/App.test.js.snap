// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders 1`] = `
<div>
  <div>
    <h1
      className="logo"
    >
      Smart Portfolio
    </h1>
    <div>
      <button
        className="googleSpace btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-google fa-w-16 "
          data-icon="google"
          data-prefix="fab"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 488 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
            Login with Google
      </button>
      <div>
        <a
          href="/Login"
          onClick={[Function]}
        >
          <button
            className="space btn btn-outline-info"
            disabled={false}
            type="button"
          >
            Smart Portfolio Login
          </button>
        </a>
        <a
          href="/signup"
          onClick={[Function]}
        >
          <button
            className="space btn btn-outline-info"
            disabled={false}
            type="button"
          >
            Smart Portfolio Signup
          </button>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`LandingPage snapshot renders 1`] = `
<div>
  <h1
    className="logo"
  >
    Smart Portfolio
  </h1>
  <div>
    <button
      className="googleSpace btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-google fa-w-16 "
        data-icon="google"
        data-prefix="fab"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 488 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M488 261.8C488 403.3 391.1 504 248 504 110.8 504 0 393.2 0 256S110.8 8 248 8c66.8 0 123 24.5 166.3 64.9l-67.5 64.9C258.5 52.6 94.3 116.6 94.3 256c0 86.5 69.1 156.6 153.7 156.6 98.2 0 135-70.4 140.8-106.9H248v-85.3h236.1c2.3 12.7 3.9 24.9 3.9 41.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
          Login with Google
    </button>
    <div>
      <a
        href="/Login"
        onClick={[Function]}
      >
        <button
          className="space btn btn-outline-info"
          disabled={false}
          type="button"
        >
          Smart Portfolio Login
        </button>
      </a>
      <a
        href="/signup"
        onClick={[Function]}
      >
        <button
          className="space btn btn-outline-info"
          disabled={false}
          type="button"
        >
          Smart Portfolio Signup
        </button>
      </a>
    </div>
  </div>
</div>
`;
